### ISSUE TITLE
Implement TypeVar bounds with an intersection bound

### ISSUE THREAD

**Issue by graingert** [python/mypy#2702]
For type variables that extend multiple interfaces

`<T extends I1 & I2>`

**Comment by gvanrossum** [comment]
Can you describe your use case for this feature, with some examples?

It sounds related to the [Intersection](https://github.com/python/typing/issues/213) proposed in the [typing](https://github.com/python/typing) repo.

In general a new feature like this should probably be discussed in the typing repo first, since it would require changes to the typing.py module (which is gated by stdlib releases).

**Comment by ilevkivskyi** [comment]
@gvanrossum Assuming the original syntax is taken from Java, this is what I proposed in the issue you mentioned, ``T = TypeVar('T', bound=Intersection[t1, t2])``. As I mentioned in that issue, the most obvious use case is when types ``t1``, ``t2`` are ABCs (similar to interfaces in Java).

**Comment by graingert** [comment]
ah yes, this is just https://github.com/python/typing/issues/213 I was searching under TypeVar

**Comment by gvanrossum** [comment]
OK, closing in favor of #213.
